cmake/modules/RootNewMacros.cmake

```diff
@@ -1449,4 +1449,34 @@ function(find_python_module module)
    set(PY_${module_upper}_FOUND ${PY_${module_upper}_FOUND} PARENT_SCOPE)
 endfunction()
 
+#----------------------------------------------------------------------------
+# Generate headers files containing the command line options help
+# The first argument pythonInput is the path of the python argparse file for this command
+# The second argument output is the of path/name of the output file
+# The third argument is the name of the target that should be linked to the generated
+# library( the executable that includes it or the library that uses it)
+#----------------------------------------------------------------------------
+function(generateHeaders pythonInput output target)
+     add_custom_command(OUTPUT ${output}
+          DEPENDS ${pythonInput} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py
+          COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py 
+                                       ${pythonInput}
+                                       ${output}
+     )
+     target_sources(${target} PRIVATE ${output})
```

Comment:
I believe that the problem is here. CMake is not understanding that `${target}` needs to wait for the custom command to run just from this call to `target_sources()`. We need to use something similar to what is used for the dictionaries. I can reproduce the problem others had on Linux by using ninja as generator. We can add one build with ninja to the PRs if needed for this. (You can also just use `@phsft-bot build just on ubuntu18/gcc7 with flags -G Ninja` or similar).